transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -151,16 +152,21 @@ public final ChannelPipeline addFirst(String name, ChannelHandler handler) {
         return addFirst(null, name, handler);
     }
 
-    @Override
-    public final ChannelPipeline addFirst(EventExecutorGroup group, String name, ChannelHandler handler) {
+    private ChannelPipeline internalAdd(EventExecutorGroup group, String name, ChannelHandler handler, String baseName, BiConsumer<AbstractChannelHandlerContext,AbstractChannelHandlerContext> addMethod){
```

Comment:
4.1 can not use `BiConsumer` as its java8.

Reply:
thanks for your review, I have replaced it with an enum(the strategy pattern may be too heavy for it). 

what do you think about this PR change idea, it is probably not required, just cleaner for me.